be/src/exprs/vectorized/array_functions.tpp

```diff
@@ -331,4 +331,167 @@ private:
         }
     }
 };
+
+class ArrayJoin {
+public:
+    static ColumnPtr process(FunctionContext* ctx, const Columns& columns) {
+        DCHECK_GE(columns.size(), 2);
+
+        size_t chunk_size = columns[0]->size();
+
+        if (columns[0]->only_null()) {
+            return ColumnHelper::create_const_null_column(chunk_size);
+        }
+
+        ColumnPtr src_column = ColumnHelper::unpack_and_duplicate_const_column(chunk_size, columns[0]);
+        Slice sep = columns[1]->get(0).get_slice();
```

Comment:
If the second and third arguments of array_join are only be const, we need to check it in FE，otherwise, we need to handle it for non-const  arguments.

Reply:
Whether it is const has no effect on IColumn::get(0)

Reply:
Fixed